R/FilterStates.R

```diff
@@ -65,55 +47,74 @@ FilterStates <- R6::R6Class( # nolint
     initialize = function(data,
                           data_reactive = function(sid = "") NULL,
                           dataname,
-                          datalabel = character(0)) {
-      checkmate::assert_function(data_reactive, args = "sid")
+                          datalabel = NULL) {
       checkmate::assert_string(dataname)
-      checkmate::assert_character(datalabel, max.len = 1, any.missing = FALSE)
+      logger::log_trace("Instantiating { class(self)[1] }, dataname: { dataname }")
+      checkmate::assert_function(data_reactive, args = "sid")
+      checkmate::assert_string(datalabel, na.ok = FALSE, null.ok = TRUE)
 
       private$dataname <- dataname
       private$datalabel <- datalabel
+      private$dataname_prefixed <- dataname
       private$data <- data
       private$data_reactive <- data_reactive
+      private$state_list <- reactiveVal()
 
       logger::log_trace("Instantiated { class(self)[1] }, dataname: { private$dataname }")
       invisible(self)
     },
 
-    #' @description
-    #' Returns the label of the dataset.
-    #'
-    #' @return `character(1)` the data label
-    #'
-    get_datalabel = function() {
-      private$datalabel
-    },
-
     #' @description
     #' Returns a formatted string representing this `FilterStates` object.
     #'
     #' @param indent (`numeric(1)`) the number of spaces prepended to each line of the output
     #'
     #' @return `character(1)` the formatted string
     #'
-    format = function(indent) {
-      sprintf(
-        paste(
-          "%sThis is an instance of an abstract class.",
-          "Use child class constructors to instantiate objects."
-        ),
-        paste(rep(" ", indent), collapse = "")
+    format = function(indent = 0) {
+      checkmate::assert_number(indent, finite = TRUE, lower = 0)
+
+      formatted_states <- vapply(
+        private$state_list_get(), function(state) state$format(indent = indent),
+        USE.NAMES = FALSE, FUN.VALUE = character(1)
       )
+      paste(formatted_states, collapse = "\n")
     },
 
     #' @description
     #' Filter call
     #'
-    #' Builds \emph{subset expression} from condition calls stored in `FilterState`
-    #' objects selection. The `lhs` of the expression is `private$dataname`.
-    #' The `rhs` is a call to `self$get_fun()` with `private$dataname`
-    #' as argument and a list of condition calls from `FilterState` objects
-    #' stored in `private$state_list`.
-    #' If no filters are applied, `NULL` is returned to avoid no-op calls such as `x <- x`.
+    #' Builds \emph{subset expression} from condition calls generated by `FilterState`.
+    #' The `lhs` of the expression is a `dataname_prefixed`, where word prefixed refers to
+    #' situation when call is using some slots of the original data, for example `dataname[[slot]]`.
```

Comment:
These are not "slots" in OOP sense, please avoid misleading verbiage. Slots of an MAE are: ExperimentList, colData, sampleMap, drops, and metadata.